### PR TITLE
cpu/atmega2560: reworked UART (from #5026)

### PR DESCRIPTION
**Pull request by kYc0o**
Takes #5026 and adds small fixes.

### CODE REVIEW

**Comment by kYc0o** [comment]
@aabadie today it's gonna be the day? ;)

File: boards/arduino-mega2560/include/periph_conf.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /*
- * Copyright (C) 2014 Freie Universität Berlin, Hinnerk van Bruinehsen
+ * Copyright (C) 2014-2016 Freie Universität Berlin
```
Comment (kaspar030) [inline boards/arduino-mega2560/include/periph_conf.h]: @kYc0o is that copyright correct? (shouldn't INRIA?)

Comment (kYc0o) [inline boards/arduino-mega2560/include/periph_conf.h]: Actually this code comes from Hauke, I didn't modify anything on this so I think is ok

Comment (kaspar030) [inline boards/arduino-mega2560/include/periph_conf.h]: ok, perfect! :)

**Comment by aabadie** [comment]
@kYc0o, code changes looks good to me but I don't have a mega2560 for testing.

**Comment by kYc0o** [comment]
@adjih doesn't have one? Or @jia200x ?

**Comment by kaspar030** [comment]
Ive got one but little time. would basic shell testing confirm that this is working?

-----Original Message-----

**Comment by kYc0o** [comment]
Somehow yes, but of you can run the test 'periph_uart'  and check for the second uart it would be ok

**Comment by aabadie** [comment]
@kYc0o, yes @adjih have some but they are used in custom setup for IoT-Lab.

**Comment by aabadie** [comment]
@kYc0o, I just gave it a try. The UART0 works.

But I tried to configure another UART and didn't succeed. I based my work by simply adding the bare minimum for UART communication in the hello example:
* in the Makefile, I added:
```
USEMODULE += xtimer
USEMODULE += periph_uart
```
* in the main.c, I added:
```c
#include "periph/uart.h"
#include "xtimer.h"
```
and, after the initial printf:
```c
uart_init(2, 115200, NULL, (void*)NULL);
    for (;;) {
        uart_write(2, (uint8_t*)"test\n", 5);
        xtimer_sleep(1000);
    }
```
but then the build fails with the error:
```
avr-gcc: error: <path_to_src>/RIOT/examples/hello-world/bin/arduino-mega2560/periph_uart.a: No such file or directory
<path_to_src>/RIOT/Makefile.include:259: recipe for target 'all' failed
make: *** [all] Error 1
```
Is it me or is there something broken ? Do all UARTs be supported on this board ?

**Comment by aabadie** [comment]
Ok, actually it was me, sorry :confused: 
I used the wrong Makefile option for `periph_uart`... (it should be `FEATURES_REQUIRED`)...

So for me it works like a charm.

ACK

**Comment by kYc0o** [comment]
@aabadie can you put the flag "ready for ci" to make murdock work?

**Comment by kYc0o** [comment]
I saw some ACK so go!

**Pull request merged**